test/util.spec.js

```diff
@@ -14,162 +16,108 @@ const fixtureBlockHeader = helpers.headerFromHexBlock(fixtureBlockHex)
-        timestamp: 1503851731,
-        bits: 402734313,
-        nonce: 3911763601
-      })
-      verifyCid(
-        dagNode,
-        '562090bd49e9fae063aa4db26ae434212157c4c72e16492aac000000000000000000',
-        done)
+    const dagNode = IpldBitcoin.util.deserialize(segwitBlockHeader)
+    verifyBlock(dagNode, {
+      version: 536870914,
+      prevHash: '92f0d678374dbb0a205345d38f35be782412207bbdaa71000000000000000000',
+      merkleRoot: '99e3557bb520c3d45d6eb6ee18f93b3665bf4c8d9747200db4292fdbacc278c3',
+      timestamp: 1503851731,
+      bits: 402734313,
+      nonce: 3911763601
     })
   })
 
-  it('should deserialize Segwit correctly (c)', (done) => {
+  it('should deserialize Segwit correctly (c)', () => {
     const segwitBlockHex = loadFixture('test/fixtures/segwit3.hex')
     const segwitBlockHeader = helpers.headerFromHexBlock(segwitBlockHex)
-    IpldBitcoin.util.deserialize(segwitBlockHeader, (err, dagNode) => {
-      expect(err).to.not.exist()
-      verifyBlock(dagNode, {
-        version: 536870912,
-        prevHash: '92fed79ebe58e1604dc08037488567c0881e1ae6a67831010000000000000000',
-        merkleRoot: '654f3617284e0c0f71baeaea9f54e337645550832b63de3dce4b66b2fbb27309',
-        timestamp: 1503848099,
-        bits: 402734313,
-        nonce: 2945767029
-      })
-      verifyCid(
-        dagNode,
-        '56205bd62fe2f3fe376ff2645681bebe7d12b64d9446ef72d1000000000000000000',
-        done)
+    const dagNode = IpldBitcoin.util.deserialize(segwitBlockHeader)
+    verifyBlock(dagNode, {
+      version: 536870912,
+      prevHash: '92fed79ebe58e1604dc08037488567c0881e1ae6a67831010000000000000000',
+      merkleRoot: '654f3617284e0c0f71baeaea9f54e337645550832b63de3dce4b66b2fbb27309',
+      timestamp: 1503848099,
+      bits: 402734313,
+      nonce: 2945767029
     })
   })
 
-  it('should error on an invalid block', (done) => {
+  it('should error on an invalid block', () => {
     const invalidBlock = Buffer.from('abcdef', 'hex')
-    IpldBitcoin.util.deserialize(invalidBlock, (err, dagNode) => {
-      expect(dagNode).to.not.exist()
-      expect(err).to.be.an('error')
-      done()
-    })
+    expect(() => {
+      IpldBitcoin.util.deserialize(invalidBlock)
+    }).to.throw('Bitcoin block header needs to be 80 bytes')
   })
 })
 
 describe('IPLD format util API serialize()', () => {
-  it('should round-trip (de)serialization correctly', (done) => {
-    IpldBitcoin.util.deserialize(fixtureBlockHeader, (err, dagNode) => {
-      expect(err).to.not.exist()
-      IpldBitcoin.util.serialize(dagNode, (err, binaryBlob) => {
-        expect(err).to.not.exist()
-        expect(binaryBlob).to.deep.equal(fixtureBlockHeader)
-        done()
-      })
-    })
+  it('should round-trip (de)serialization correctly', () => {
+    const dagNode = IpldBitcoin.util.deserialize(fixtureBlockHeader)
+    const binaryBlob = IpldBitcoin.util.serialize(dagNode)
+    expect(binaryBlob).to.deep.equal(fixtureBlockHeader)
```

Comment:
Is chai good with proper `Buffer` comparisons? Or would it be better go with a more direct comparison using `Buffer` internals: `binaryBlob.equals(fixtureBlockHeader)`

Reply:
So far it worked well for me.